src/apps/bristleback_apps/aanderaa/user_code/user_code.cpp

```diff
@@ -162,36 +189,52 @@ void setup(void) {
   configASSERT(userConfigurationPartition);
   userConfigurationPartition->getConfig("currentAggPeriodMin", strlen("currentAggPeriodMin"),
                                         current_agg_period_min);
+  userConfigurationPartition->getConfig("nSkipReadings", strlen("nSkipReadings"),
+                                        n_skip_readings);
+  userConfigurationPartition->getConfig("readingIntervalMs", strlen("readingIntervalMs"),
+                                        reading_interval_ms);
+  userConfigurationPartition->getConfig("payloadWdToS", strlen("payloadWdToS"),
+                                        payload_wd_to_s);
+  userConfigurationPartition->getConfig("plUartBaudRate", strlen("plUartBaudRate"),
+                                        baud_rate);
+
+  max_readings_in_agg = (((uint64_t)CURRENT_AGG_PERIOD_MS / reading_interval_ms) + N_SAMPLES_PAD);
 
   aanderaaTopicStrLen = createAanderaaDataTopic();
   // Allocate memory for pressure data buffer.
   for (uint8_t i = 0; i < NUM_PARAMS_TO_AGG; i++) {
-    current_data[i].initBuffer(MAX_CURRENT_SAMPLES);
+    current_data[i].initBuffer(max_readings_in_agg);
   }
   stats_print_buffer = static_cast<char *>(
       pvPortMalloc(sizeof(char) * 1500)); // Trial and error, should be plenty of space
   // Initialize our LineParser, which will allocated any needed memory for parsing.
   parser.init();
   // Setup the UART – the on-board serial driver that talks to the RS232 transceiver.
   PLUART::init(USER_TASK_PRIORITY);
-  // Baud set per expected baud rate of the sensor.
-#ifdef AANDERAA_BLUE
-  PLUART::setBaud(115200);
-#else
-  PLUART::setBaud(9600);
-#endif
+  // Baud set per configuration.
+  PLUART::setBaud(baud_rate);
   // Set a line termination character per protocol of the sensor.
   PLUART::setTerminationCharacter('\n');
   // Turn on the UART.
   PLUART::enable();
-  // Enable the input to the Vout power supply.
+  // Enable the input to the Vout power supply and ensure PL buck is disabled.
   IOWrite(&BB_VBUS_EN, 0);
-  // ensure Vbus stable before enable Vout with a 5ms delay.
-  vTaskDelay(pdMS_TO_TICKS(5));
+  IOWrite(&BB_PL_BUCK_EN, 1);
+  // ensure Vbus stable before enable Vout with a 1.5s delay.
+  vTaskDelay(pdMS_TO_TICKS(1500));
```

Comment:
I'm curious about the change from a 5ms delay to 1500ms. In September, Michael and I were fixing I2C bugs, and he did some oscilloscope validations to see how long of a delay we needed in order for the 3v3 rail to stabilize. Slack conversation [here](https://sofarocean.slack.com/archives/C05J96Q6QLX/p1695754718673349?thread_ts=1695750119.795059&cid=C05J96Q6QLX). Does it take longer with the Aanderaa sensor attached? Or did you find you needed to increase it this much based on experience testing the sensor in real conditions?

Reply:
I had originally changed the delay to `AANDERAA_RESET_TIME_MS` (500ms) per the note Russ added in the definition of that constant. Intent is to ensure that whatever was going on before we got here, we turned the thing off long enough to ensure it fully power cycled.
I then increased the delay to a hard-coded 1.5s because 500ms was too fast for my serial terminal to reliably capture the debug prints a few lines down. 😬 
I'll revert to `AANDERAA_RESET_TIME_MS`, which is the best-informed duration here imo.

Reply:
Ah! Right, yes, awesome. I remember him looking up that duration. Perfect.